pandas/tseries/index.py

```diff
@@ -77,16 +77,19 @@ def f(self):
 
             result = tslib.get_start_end_field(values, field, self.freqstr,
                                                month_kw)
+            result = self._maybe_mask_results(result, convert='float64')
+
```

Comment:
I think you can have a single ``result = self._maybe_mask_results(result, convert='float64')`` just before returning; it won't do anything to something w/o nan's anyhow (and is more clear code)